delta_processor.go

```
@@ -79,10 +77,28 @@ func (dp *deltaProcessor) process(delta *api.ClientFeaturesDelta) error {
 		}
 	}
 
-	if err := dp.repository.updateStorageWithDelta(currentFeatures, segments); err != nil {
```

I've chosen to discard this error handling branch because I think its wrong. This can only occur if writing to the persistent storage fails in the v5 implementation. The data store is still updated, which means feature flags are still working correctly. I think this is especially problematic with custom implementations of the data store where we don't control the error paths